core/Plugin.php

```diff
@@ -461,6 +463,42 @@ public function getMissingDependenciesAsString($piwikVersion = null)
         ));
     }
 
+    /**
+     * Schedules re-archiving of this plugin's reports from when this plugin was last
+     * deactivated to now. If the last time core:archive was run is earlier than the
+     * plugin's last deactivation time, then we use that time instead.
+     *
+     * Note: this only works for CLI archiving setups.
+     *
+     * Note: the time frame is limited by the `[General] rearchive_reports_in_past_last_n_months`
+     * INI config value.
+     *
+     * @throws \DI\DependencyException
+     * @throws \DI\NotFoundException
+     */
+    public function schedulePluginReArchiving()
+    {
+        $lastDeactivationTime = $this->getPluginLastDeactivationTime();
+
+        $dateTime = null;
+
+        $lastCronArchiveTime = (int) Option::get(CronArchive::OPTION_ARCHIVING_FINISHED_TS);
+        if (empty($lastCronArchiveTime)) {
+            $dateTime = $lastDeactivationTime;
+        } else if (empty($lastDeactivationTime)) {
+            $dateTime = Date::factory($lastCronArchiveTime);
```

Comment:
Shouldn't `$dateTime` be `null` if the plugin had not been deactivated before, so it triggers archiving for all dates in the past? 🤔

Reply:
@diosmosis correct me if I'm wrong but the rearchive method should then limit it to say max 6 months (or whatever value is configured)

Reply:
it would be limited here: https://github.com/matomo-org/matomo/blob/4.0.0-rc4/core/Archive/ArchiveInvalidator.php#L470-L472

Reply:
@diosmosis did you have a look at my comment? Still thinking using `$lastCronArchiveTime` in this case seems wrong.

Reply:
@sgiehl it should then be limited see link above or not? also if `lastCronArchiveTime` is empty then it would have went in the first `if`

Reply:
But `$dateTime` is set to the last archiving time, so the reports would be built from that date on, or not?

Reply:
I see. Assuming it last ran  today it would only rearchive today? Is this what you mean? We could set it to `null` in that case?

Reply:
@sgiehl created https://github.com/matomo-org/matomo/pull/16748 good find 👍

Reply:
I think it's not expected for the plugin deactivation time to be unset except in the case of installing a plugin (ie, the normal workflow is deactivate plugin, later activate, which triggers this code). In that case we can just set it to null?